### PR TITLE
B13 - unit_test fixes

### PR DESCRIPTION
**Pull request by vinniefalco**
This is needed for NuDB

Reviewers: @nbougalis @HowardHinnant

### CODE REVIEW

File: extras/beast/unit_test/dstream.hpp [hunk 1]
```diff
@@ -8,9 +8,7 @@
 #ifndef BEAST_UNIT_TEST_DSTREAM_HPP
 #define BEAST_UNIT_TEST_DSTREAM_HPP
 
-#include <boost/utility/base_from_member.hpp>
 #include <iostream>
-#include <memory>
 #include <ostream>
 #include <sstream>
 #include <string>
```
Comment (HowardHinnant) [inline extras/beast/unit_test/dstream.hpp]: Here's the set of headers I recommend:
````
#include <ios>
#include <memory>
#include <ostream>
#include <streambuf>
#include <string>
````
You could get away with less, but there is little motivation to do so.

**Comment by HowardHinnant** [comment]
Just one picky comment about headers.  Tests out on macOS.  👍

**Comment by vinniefalco** [comment]
How'd you come to that conclusion about the headers (and I'm not saying I disagree, just want to know the methodology)

**Comment by HowardHinnant** [comment]
I searched for `std` and for each one I hit, I noted the header that associated identifier lives in.  Then I alphabetized the list.

File: extras/beast/unit_test/dstream.hpp (hunk 1 above)
Comment (vinniefalco) [inline extras/beast/unit_test/dstream.hpp]: WillFix

File: extras/beast/unit_test/dstream.hpp [hunk 2]
```diff
@@ -79,66 +81,42 @@ class dstream_buf
...
-        return 0;
-    }
-};
-
-#endif
-
 } // detail
 
-/// A std::ostream that redirects output to the debugger if attached.
+/// A std::ostream that redirects output to the MSVC debugger if attached.
```
Comment (nbougalis) [inline extras/beast/unit_test/dstream.hpp]: "if one is attached when the stream is created"

File: extras/beast/unit_test/dstream.hpp [hunk 3]
```diff
@@ -79,66 +81,42 @@ class dstream_buf
...
     }
 };
 
 using dstream = basic_dstream<char>;
 using dwstream = basic_dstream<wchar_t>;
 
-} // test
+#else
+
+using dstream = std::ostream&;
```
Comment (nbougalis) [inline extras/beast/unit_test/dstream.hpp]: Why a reference?

**Comment by nbougalis** [comment]
Left one pico-nit about a comment. :+1:

**Comment by vinniefalco** [comment]
@nbougalis willfix / didfix

**Pull request merged**